[COMMONSSITE-156] Reviewing Release Plug-in post Imaging 1.0-alpha3 release

kinow opened the pull request:
Had a few hiccups during the last release of Imaging, so I'm sending this PR to the Release Plug-in, and will prepare a SVN patch for the Commons CMS site.

- [x] typos
- [x] fixing HTML tags (e.g. adding `lang="en"` as it's recommended for accessibility)
- [x] remove unused methods in the plug-in (not deprecated, but actually not used anywhere within this code base)
- [x] fixing an issue found in the imaging release when running `mvn site` due to a change in the parent
- [x] use HTTPS when possible
- [ ] fixing RAT, pending the 0.14 release of their maven plug-in (<https://github.com/apache/commons-imaging/blob/de0dda58cdabb5c22c6c2596dfa085262ce64840/pom.xml#L29-L32>)
  * <https://issues.apache.org/jira/browse/RAT-297>
  * <https://issues.apache.org/jira/browse/MSHARED-1049>

The last part about RAT is a nice to have. At the moment the plug-in is failing to run when you run `mvn site`, but it doesn't break the build.

The reason is that the RAT plug-in had been using a deprecated call for years, that was eventually removed from the Maven toolkit. Once they release the 0.14 release with the fix it should be fixed. So in other words, at the moment RAT plug-in is not working in the Release Plug-in, but you are able to run `mvn apache-rat:check` and `mvn apache-rat:rat`.

I think this can be merged as-is, and the RAT issue moved to its own issue (or we can downgrade to parent 52)

raboof commented:
@kinow I suspect we may have missed that this was ready for review because it was still assigned to you - do you think the changes in this PR are still relevant?

kinow commented:
Hi @raboof , I'm sorry but I don't know if that's still relevant. It could be that the content hsa been updated addressing the points I updated here. I did that whilst I released an alpha version of Imaging, some time ago. I think it was probably updated by others, and we can close this -- others might update the docs anyway once they find missing instructions (which is what I think I updated here).
